hexkit/providers/akafka.py

```diff
@@ -86,3 +104,112 @@ def publish(self, *, payload: JsonObject, type_: str, key: str, topic: str) -> N
+
+        Args:
+            service_name (str):
+                The name of the (micro-)service from which messages are published.
+            client_suffix (str):
+                String that uniquely this instance across all instances of this service.
+                Will create a globally unique Kafka client IDidentifier by concatenating
+                the service_name and the client_suffix.
+            kafka_servers (list[str]):
+                List of connection strings pointing to the kafka brokers.
+            translator (EventSubscriberProto):
+                The translator that translates between the protocol (mentioned in the
+                type annotation) and an application-specific port
+                (according to the triple hexagonal architecture).
+            kafka_consumer_cls:
+                Overwrite the used Kafka Producer class. Only intented for unit testing.
+        """
+
+        client_id = generate_client_id(service_name, client_suffix)
+        self._translator = translator
+        topics = self._translator.topics_of_interest
+        self._types_whitelist = translator.types_of_interest
+
+        self._consumer = kafka_consumer_cls(
+            *topics,
+            bootstrap_servers=kafka_servers,
+            client_id=client_id,
+            group_id=service_name,
+            auto_offset_reset="earliest",
+            key_deserializer=lambda event_key: event_key.decode("ascii"),
+            value_deserializer=lambda event_value: json.loads(
+                event_value.decode("ascii")
+            ),
+        )
+
+    @staticmethod
+    def _get_type(event: ConsumerRecord) -> str:
+        """Extract the event type out of an ConsumerRecord."""
+        for header in event.headers:
+            if header[0] == "type":
+                return header[1].decode("ascii")
+        raise EventTypeNotFoundError()
+
+    @staticmethod
+    def _get_event_label(event: ConsumerRecord) -> str:
+        """Get a label that identifies an event."""
+        return (
+            f"{event.topic} - {event.partition} - {event.offset} "
+            + " (topic-partition-offset)"
+        )
+
+    def _consume_event(self, event: ConsumerRecord) -> None:
+        """Consume an event by passing it down to the translator via the protocol."""
+        event_label = self._get_event_label(event)
+
+        try:
+            type_ = self._get_type(event)
+        except EventTypeNotFoundError:
+            logging.warning("Ignored an event without type: %s", event_label)
+        else:
+
+            if type_ in self._types_whitelist:
+                logging.info('Consuming event of type "%s": %s', type_, event_label)
+
+                try:
+                    # blocks until event processing is completed:
+                    self._translator.consume(
+                        payload=event.value, type_=type_, topic=event.topic
+                    )
+                except:  # pylint: disable=bare-except
```

Comment:
```suggestion
                except Exception:
```

Reply:
I think, then it complains about using to generic exceptions 😆 But let's give it a try.